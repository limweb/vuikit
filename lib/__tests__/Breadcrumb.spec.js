import { createVue, destroyVM, queryByTag, queryByTagAll, toJSON } from './_util'
import waitForUpdate from './_wait-for-update'

describe('Breadcrumb', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-breadcrumb location="/blog">
        <vk-breadcrumb-item path="/">Home</vk-breadcrumb-item>
        <vk-breadcrumb-item label="Blog" path="/blog" />
        <vk-breadcrumb-item label="Category" path="/blog/category" />
        <vk-breadcrumb-item label="Post" path="/blog/category/post" disabled />
      </vk-breadcrumb>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('triggers event change when item is clicked', done => {
    vm = createVue(`
      <vk-breadcrumb location="/">
        <vk-breadcrumb-item label="Home" path="/" />
        <vk-breadcrumb-item label="Blog" path="/blog" />
      </vk-breadcrumb>
    `)
    const breadcrumb = queryByTag(vm, 'vk-breadcrumb')
    const items = queryByTagAll(vm, 'vk-breadcrumb-item')
    const cb = jest.fn()

    breadcrumb.$on('change', cb)
    waitForUpdate(() => {
      items[1].$el.querySelector('a').click()
    }).then(() => {
      expect(cb).toHaveBeenCalledWith('/blog')
    }).then(done)
  })
})
