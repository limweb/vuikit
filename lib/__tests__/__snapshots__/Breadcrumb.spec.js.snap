exports[`Breadcrumb renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-breadcrumb",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "Home",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "Blog",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "Category",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "Post",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;
